partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/V1RexConverter.kt

```diff
@@ -0,0 +1,1077 @@
+            return rex(type, op)
+        }
+
+        // SPECIAL FORMS
+
+        /**
+         * <arg0> NOT? LIKE <arg1> ( ESCAPE <arg2>)?
+         */
+        override fun visitExprLike(node: ExprLike, ctx: Env): Rex {
+            val type = BOOL
+            // Args
+            val arg0 = visitExprCoerce(node.value, ctx)
+            val arg1 = visitExprCoerce(node.pattern, ctx)
+            val arg2 = node.escape?.let { visitExprCoerce(it, ctx) }
+            // Call Variants
+            var call = when (arg2) {
+                null -> call("like", arg0, arg1)
+                else -> call("like_escape", arg0, arg1, arg2)
+            }
+            // NOT?
+            if (node.not == true) {
+                call = negate(call)
+            }
+            return rex(type, call)
+        }
+
+        /**
+         * <arg0> NOT? BETWEEN <arg1> AND <arg2>
+         */
+        override fun visitExprBetween(node: ExprBetween, ctx: Env): Rex = plan {
+            val type = BOOL
+            // Args
+            val arg0 = visitExprCoerce(node.value, ctx)
+            val arg1 = visitExprCoerce(node.from, ctx)
+            val arg2 = visitExprCoerce(node.to, ctx)
+            // Call
+            var call = call("between", arg0, arg1, arg2)
+            // NOT?
+            if (node.not == true) {
+                call = negate(call)
+            }
+            rex(type, call)
+        }
+
+        /**
+         * <arg0> NOT? IN <arg1>
+         *
+         * SQL Spec 1999 section 8.4
+         * RVC IN IPV is equivalent to RVC = ANY IPV -> Quantified Comparison Predicate
+         * Which means:
+         * Let the expression be T in C, where C is [a1, ..., an]
+         * T in C is true iff T = a_x is true for any a_x in [a1, ...., an]
+         * T in C is false iff T = a_x is false for every a_x in [a1, ....., an ] or cardinality of the collection is 0.
+         * Otherwise, T in C is unknown.
+         *
+         */
+        override fun visitExprInCollection(node: ExprInCollection, ctx: Env): Rex {
+            val type = BOOL
+            // Args
+            val arg0 = visitExprCoerce(node.lhs, ctx)
+            val arg1 = node.rhs.accept(this, ctx) // !! don't insert scalar subquery coercions
+
+            // Call
+            var call = call("in_collection", arg0, arg1)
+            // NOT?
+            if (node.not == true) {
+                call = negate(call)
+            }
+            return rex(type, call)
+        }
+
+        /**
+         * <arg0> IS <NOT>? <type>
+         */
+        override fun visitExprIsType(node: ExprIsType, ctx: Env): Rex {
+            val type = BOOL
+            // arg
+            val arg0 = visitExprCoerce(node.value, ctx)
+            val targetType = node.type
+            var call = when (targetType.code()) {
+                // <absent types>
+                DataType.NULL -> call("is_null", arg0)
+                DataType.MISSING -> call("is_missing", arg0)
```

Comment:
Something I didn't catch from before. `ExprIsType` should be distinct from `ExprIsNull` and `ExprIsMissing` and `ExprIsUnknown`. DataType shouldn't contain the `NULL` and `MISSING` variants. 
That way, you don't need those checks in the `visitExprCast`. Since this modelling is not the main topic of this PR, this can be a follow-up if you'd like.

Reply:
Yeah, @RCHowell gave me some context. The previous sprout AST modeling for `IS` and `type` were not correct, hence why the current v1 AST modeling is not correct. The correct modeling of `IS` in the ast and plan will need some additional spec work so will be addressed in a future PR before v1 is released. I've added some TODOs in data type and `ExprIsType` for now.